mds/apis/agency_api/v0_x/vehicles.py

```diff
@@ -186,13 +195,37 @@ class DeviceEventSerializer(serializers.Serializer):
         ),
     )
 
+    def get_event(self, validated_data):
+        # We first check whether the provider uses the old or the new agency
+        # event_type(s) and event_type_reason(s).
+        # A provider uses the old version if agency_api_version == "draft" in its
+        # api_configuration.
+        provider_id = self.context["request"].user.provider_id
+        provider = models.Provider.objects.get(id=provider_id)
+        old_version = provider.api_configuration.get("agency_api_version") == "draft"
+        event_type = validated_data.get("event_type")
+        event_type_reason = validated_data.get("event_type_reason")
+        if old_version:
+            return (event_type, None)
+        else:
+            event = (event_type, event_type_reason)
+            if event not in provider_mapping.AGENCY_EVENT_TO_PROVIDER_REASON.keys():
+                # This should be avoided if possible
+                logger.exception(
+                    f"The event ({event[0]}, {event[1]}) is not in the mapping."
+                )
```

Comment:
What exception? Logging will probably complain. Do you want to raise an exception?
```python
try:
    provider_mapping.AGENCY_EVENT_TO_PROVIDER_REASON[event]
except KeyError:
    logger.exception(...)
    raise ValidationError("...")
```
Here you allow creating an event record will invalid data.